### PR TITLE
ZKUI -17: Add object lock setting page

### PR DESCRIPTION
**Pull request by ChengYanJin**
Add object-lock setting button in the bucket overview tab:
image

And object-lock setting page:
image

### CODE REVIEW

File: src/react/databrowser/buckets/ObjectLockSetting.jsx [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+      otherwise: Joi.string(),
+    },
+  ),
+});
+
+export default function ObjectLockSetting() {
+  const bucketInfo = useSelector((state: AppState) => state.s3.bucketInfo);
+  const loading = useSelector(
+    (state: AppState) => state.networkActivity.counter > 0,
+  );
```
Comment (JBWatenbergScality) [inline src/react/databrowser/buckets/ObjectLockSetting.jsx]: const isLoading = isObjectLockEnabled

File: src/react/databrowser/buckets/ObjectLockSetting.jsx [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+              tooltipWidth="28rem"
+            >
+              Object-lock
+            </F.Label>
+            <F.Label>
+              {bucketInfo?.objectLockConfiguration?.ObjectLockConfiguration
+                ?.ObjectLockEnabled || 'loading'}
+            </F.Label>
+          </F.Fieldset>
+          {isObjectLockEnabled && (
```
Comment (JBWatenbergScality) [inline src/react/databrowser/buckets/ObjectLockSetting.jsx:117]: isObjectLockEnabled === 'Enabled'

File: src/react/ui-elements/TableKeyValue2.jsx [hunk 3]
```diff
@@ -151,6 +151,12 @@ export const Value = styled.div`
   }
 `;
 
+export const GroupValues = styled.div`
+  display: flex;
+  flex: 1 1 65%;
+  justify-content: space-between;
```
Comment (JBWatenbergScality) [inline src/react/ui-elements/TableKeyValue2.jsx:157]: align-items: baseline

**Review by JBWatenbergScality — APPROVED** [review]

File: src/react/databrowser/buckets/ObjectLockSetting.jsx [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+import Joi from '@hapi/joi';
+import { joiResolver } from '@hookform/resolvers';
+import { push } from 'connected-react-router';
+import type { AppState } from '../../../types/state';
+
+const schema = Joi.object(objectLockRetentionSettingsValidationRules);
+
+export default function ObjectLockSetting() {
+  const bucketInfo = useSelector((state: AppState) => state.s3.bucketInfo);
+  const hasError = useSelector(
```
Comment (nicolas2bert) [inline src/react/databrowser/buckets/ObjectLockSetting.jsx:23]: We should handle potential errors.

File: src/react/databrowser/buckets/ObjectLockRetentionSettings.jsx [hunk 5]
```diff
@@ -6,6 +6,26 @@ import SpacedBox from '@scality/core-ui/dist/components/spacedbox/SpacedBox';
 import { spacing } from '@scality/core-ui/dist/style/theme';
 import * as F from '../../ui-elements/FormLayout';
 import React from 'react';
+import Joi from '@hapi/joi';
+
+export const objectLockRetentionSettingsValidationRules = {
+  isDefaultRetentionEnabled: Joi.boolean(),
```
Comment (nicolas2bert) [inline src/react/databrowser/buckets/ObjectLockRetentionSettings.jsx]: Let's force the default value of `isDefaultRetentionEnabled` to `false` here.

File: src/react/databrowser/buckets/ObjectLockSetting.jsx [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+    (state: AppState) =>
+      !!state.uiErrors.errorMsg && state.uiErrors.errorType === 'byComponent',
+  );
+  const objectLockEnabled =
+    bucketInfo?.objectLockConfiguration.ObjectLockEnabled;
+  const isObjectLockEnabled = objectLockEnabled === 'Enabled';
+
+  // we may not be able to benefit from the defaultValues if we directly access the URL
+  const isDefaultRetentionEnabled = bucketInfo?.objectLockConfiguration.Rule
```
Comment (nicolas2bert) [inline src/react/databrowser/buckets/ObjectLockSetting.jsx]: We could define a function that takes bucket info and converts it to a "form" object.
And at the bucket creation, do the opposite takes "form infos" and converts them to bucket info.
Something like this: https://github.com/scality/zenko-ui/blob/development/1.1/src/react/backend/location/utils.js#L58

**Review by nicolas2bert — CHANGES_REQUESTED** [review]
Missing component tests.

**Comment by nicolas2bert** [comment]
Also, it should target development/1.0 branch. :)

**Review by nicolas2bert — APPROVED** [review]
LGTM, please squash the commits. 
Should @Cuervino try/manual test the feature since new changes have been made?

**Pull request merged**